atom/browser/net/atom_network_delegate.cc

```diff
@@ -226,6 +227,11 @@ void AtomNetworkDelegate::SetResponseListenerInIO(
     response_listeners_[type] = { patterns, callback };
 }
 
+void AtomNetworkDelegate::SetDevToolsNetworkEmulationClientId(
+    const std::string& client_id) {
+  client_id_ = client_id;
```

Comment:
`SetDevToolsNetworkEmulationClientId` is called in main thread while `OnBeforeURLRequest` in IO thread, we should protect `client_id_` with a mutex.